src/lib.rs

```diff
@@ -565,9 +562,13 @@ impl<T, U> These<T, U> {
     /// ```
     ///
     /// [`This`]: enum.These.html#variant.This
-    pub fn is_this(self) -> bool
+    pub fn is_this(&self) -> bool
     {
-        self.this_option().is_some()
+        if let These::This(_) = self {
```

Comment:
Ah ya, this is relatively new syntax to me! I need to keep it in mind in the future :)